### PR TITLE
Issue206 clean up dicstrings and add doctests

### PR DESCRIPTION
**Pull request by jonbergh**
See #206 for comments.

### CODE REVIEW

**Comment by mortenwh** [comment]
@akorosov - could you check what's wrong with appveyor? It seems to be a installation issue, and not a bug in the present pull request...

File: nansat/mappers/mapper_netcdf_cf.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def _time_count_to_np_datetime64(self, time_count, time_units=None):
             raise Exception('Check time units..')
         return tt
 
-    def _band_list(self, gdal_dataset, gdal_metadata, netcdf_dim={}, bands=[], *args, **kwargs):
```
Comment (akorosov) [inline nansat/mappers/mapper_netcdf_cf.py]: Why is that removed? It seems that the 'revert commit' 53e91c7 does something harmful here and maybe in some other files. Can you double check, @mortenwh ?

File: docs/source/notebooks/nansat-introduction.ipynb [hunk 2]
```diff
@@ -129,7 +129,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "n.write_map('map.png', pltshow=True)"
+    "n.write_figure('map.png', pltshow=True)"
```
Comment (akorosov) [inline docs/source/notebooks/nansat-introduction.ipynb]: Originally it was `write_map`, bu now we have moved it to `tools.write_domain_map`. And `Nansat.write_figure` doesn't have pltshow, option.

File: nansat/nansat.py [hunk 3]
```diff
@@ -412,12 +414,13 @@ def resize(self, factor=None, width=None, height=None, pixelsize=None, resample_
...
             current pixelsize to the desired pixelsize.
         resample_alg : int (GDALResampleAlg), optional
-               -1 : Average (default),
-                0 : NearestNeighbour
-                1 : Bilinear,
-                2 : Cubic,
-                3 : CubicSpline,
-                4 : Lancoz
+
+            - 1 : Average (default),
```
Comment (akorosov) [inline nansat/nansat.py]: Oops! -1 (minus one) is Average. Not - 1 (dash one). Should be : `- -1: Average (default)`

File: nansat/mappers/mapper_netcdf_cf.py (hunk 1 above)
Comment (mortenwh) [reply nansat/mappers/mapper_netcdf_cf.py]: this is strange... I did an erroneous merge with master, undid it, and merged with develop instead... So it hasn't worked as intended?

File: docs/source/notebooks/nansat-introduction.ipynb (hunk 2 above)
Comment (mortenwh) [reply docs/source/notebooks/nansat-introduction.ipynb]: what does this mean? Is the code correct as it is, or not?

File: nansat/mappers/mapper_netcdf_cf.py (hunk 1 above)
Comment (mortenwh) [reply nansat/mappers/mapper_netcdf_cf.py]: Both versions a478ab6398da095f90224c84fbe898a631e91ab0 and 53e91c7febf760d8f59f25e8cd6c170b311926c5 lack `*args` and `**kwargs`. None of these are used in the function so maybe not needed?

There is no difference between a478ab6398da095f90224c84fbe898a631e91ab0 and 53e91c7febf760d8f59f25e8cd6c170b311926c5. But, when I look at the blame in github, it seems `*args` and `**kwargs` were added in a commit in master. Since it didn't exist in issue206_clean_up_dicstrings_and_add_doctests it has been removed when the merge with master was reverted. Since the history is kept, this removal happens after the addition in master, and therefore it is not added when we later merge with develop (which has the same changes from master) instead.... Does this make sense? 

In other words: Since the merge still remains in the history, trying to merge that same branch in again later won't include any commits from the merging branch prior to the reverted merge. 

I `git revert 53e91c7`...

**Comment by mortenwh** [comment]
This should be ok now - @akorosov can you have another look?
